framework/generated/vulkan_generators/gencode.py

```diff
@@ -873,11 +873,28 @@ def gen_target(args):
 
     reg = Registry(gen, options)
 
+    ## @note We parse vk.xml to an in-memory element tree and then extract the info we need
+    ## from that into the Registry object once per output file we generate rather than once
+    ## per run of the top-level generation script.
     start_timer(args.time)
     tree = etree.parse(args.registry)
     gen.VIDEO_TREE = etree.parse(args.video)
     end_timer(args.time, '* Time to make ElementTree =')
 
+    # Apply any temporary patches to the xml to allow correct code generation
+    # from it. These should be reviewed and removed when the xml is fixed upstream.
+    start_timer(args.time)
+    for type in tree.iter('type'):
+        # Workaround 1.2.264 VkFrameBoundaryEXT.pTag lacking len field:
+        # <https://github.com/KhronosGroup/Vulkan-Docs/pull/2240>
+        if type.get('name') == 'VkFrameBoundaryEXT' and type.get('category') == 'struct':
+            for member in tree.iter('member'):
+                name = member.find('name')
+                if(name != None and name.text == 'pTag'):
+                    member.set('len', 'tagSize')
+                    ## @todo Early-out the top-level find as all patches are complete (throw?)
+    end_timer(args.time, '* Time to patch ElementTree =')
+
```

Comment:
I'd like eye on this please. This changes the in-memory representation of vk.xml to patch an issue in Vulkan Headers that came in at 1.3.264 and has not been fixed yet. It lets our automated codegen handle VkFrameBoundaryEXT correctly for encode and decode.

Reply:
LGTM. You _could_ wrap it all up in a single XPath query, but I'm not sure if that is any more clear:
```
    if ptag_member := tree.find('types/type[@name="VkFrameBoundaryEXT"]/member[name="pTag"]'):
        ptag_member.set('len', 'tagSize')
```